pkg/platform/kube/functionres/lazy.go

```diff
@@ -1898,6 +1898,11 @@ func (lc *lazyClient) getFunctionVolumeAndMounts(function *nuclioio.NuclioFuncti
 	var volumes []v1.Volume
 	var volumeMounts []v1.VolumeMount
 
+	// using maps to eliminate duplicates where user use the same volume to be mounted by multiple volume mounts
+	// e.g.: volume config-map X, mount it twice to different paths
+	volumeName := map[string]v1.Volume{}
+	volumeToVolumeMounts := map[string][]v1.VolumeMount{}
```

Comment:
Naming inconsistency
the 2nd map dos not map volume to volume-mounts but volume name to volume mount (and the first volume name to volume)
IMO 2 options,
option 1:
```
	volumeMap := map[string]v1.Volume{}
	volumeMountsMap := map[string][]v1.VolumeMount{}
```
option 2:
```
	volumeNameToVolume := map[string]v1.Volume{}
	volumeNameToVolumeMounts := map[string][]v1.VolumeMount{}
```